### PR TITLE
[MIK-167]Don't delete input text when query fails

### PR DESCRIPTION
**Pull request by Lredigonda**

### CODE REVIEW

File: apps/interactor/src/state/slices/narrationSlice.ts [hunk 1]
```diff
@@ -101,6 +101,7 @@ const narrationSlice = createSlice({
       toast.error(action.payload || 'Error querying the AI')
       const interaction =
         state.interactions[response?.parentInteractionId || '']
+      state.input.text = interaction?.query || ''
```
Comment (miku448) [inline apps/interactor/src/state/slices/narrationSlice.ts:104]: this should only be done if there as state.input.text before line 57 `state.input.text = ''`
this is because the interaction?.query could also be the prompt for a scenario change or an item usage.

What I suggest to do it to set `state.input.text = ''` only when the interaction is successful, let me know if that works

Comment (Lredigonda) [reply apps/interactor/src/state/slices/narrationSlice.ts:104]: it doesn't work, i put console.log in each function and got this:
image
interactionSuccess run before interactionFailure and it cleans the input.

File: apps/interactor/src/state/listeners/interaction.tsx [hunk 2]
```diff
@@ -103,12 +103,15 @@ const interactionEffect = async (
...
-      )
+      const text = result.get('text')
+      if (text && text?.length > 0) {
+        dispatch(
+          interactionSuccess({
+            ...currentResponseState,
+            completed: false,
+          })
+        )
+      }
```
Comment (miku448) [inline apps/interactor/src/state/listeners/interaction.tsx:114]: this delays the emotion display and should not be done
you should focus only delete the input when the interactionSuccess is called with `completed: true` like it's done in the line below. you shouldn't have to modify this file

File: apps/interactor/src/state/slices/narrationSlice.ts [hunk 3]
```diff
@@ -166,9 +165,15 @@ const narrationSlice = createSlice({
         if (action.payload.nextScene) {
           response.nextScene = action.payload.nextScene
         }
+
+        if (!response.fetching) {
+          state.input.text = ''
+        }
```
Comment (miku448) [inline apps/interactor/src/state/slices/narrationSlice.ts:171]: why is this necessary?

Comment (Lredigonda) [reply apps/interactor/src/state/slices/narrationSlice.ts:171]: because the text persists in the input until the stream ends

Comment (miku448) [reply apps/interactor/src/state/slices/narrationSlice.ts:171]: I mean, why is not line 176 enough?
